context_header.go

```diff
@@ -36,31 +36,56 @@ type ContextWithHeaders interface {
 	SetResponseHeaders(map[string]string)
 }
 
+type headersContextKey int
```

Comment:
I saw it, but the existing constant was explicitly initialized to `1`, so there is a chance someone might crease a clash. But if we can move the new constant to the same file and use the `iota` pattern, then it's safe.